Jenkinsfile

```diff
@@ -0,0 +1,84 @@
+pipeline {
+    agent { 
+        docker { 
+            image 'argo.registry:5000/epel-7-mgo' 
+            args '-u jenkins:jenkins'
+        }
+    }
+    options { checkoutToSubdirectory('argo-messaging') }
+    environment {
+        PROJECT_DIR='argo-messaging'
+        GOPATH="${WORKSPACE}/go"
+        GIT_COMMIT=sh(script: 'cd ${WORKSPACE}/argo-messaging && git log -1 --format="%H"',returnStdout: true).trim()
```

Comment:
Shouldn't this one be $PROJECT_DIR as well ?

Reply:
I tried it but for some reason it would interpret the ${PROJECT_DIR} variable as empty string ' '.
I can investigate it further

Reply:
Please do it would be useful to make this one generic, 
I will merge this version to try it out